v2/context/context.go

```diff
@@ -57,6 +57,48 @@ func TopicFrom(ctx context.Context) string {
 	return ""
 }
 
+// Opaque key type used to store topic partition
+type topicPartitionKeyType struct{}
```

Comment:
question: compared to the other types/functions in this package, this seems very Kafka specific. I'm assuming you have this in this package to influence per message handling vs in the protocol? IIRC, we have other protocol implementations allowing per-message send/receive properties in the protocol w/out changing this package. Can you please double-check?

Reply:
My initial reason for putting these properties in this package is that the sarama binding might leverage them. Like you said it's only for the kafka, to avoid unnecessary reference confusion, I have removed them from here.